lib/ds/heap.rb

```diff
@@ -0,0 +1,46 @@
+require_relative './doubly_linked_list'
+
+class Heap < DoublyLinkedList
+  
+  def insert_smallest_to_highest(data)
```

Comment:
This method expects a struct or an object that exposes 'value' and 'frequency' properties. It is utilized to sort each added node within the linked list based on ascending frequency, from the smallest to the highest frequency

So the linked list will have something like this:
![image](https://github.com/ImMPrada/short-ds-challenges/assets/26731448/f3eec760-40c4-4a36-96d6-7cedac41d8d4)